include/unifex/any_unique.hpp

```diff
@@ -450,13 +450,21 @@ class _byref<CPOs...>::type
     (requires (!same_as<Concrete const, type const>))
   /*implicit*/ type(Concrete& impl)
     : vtable_(vtable_holder_t::template create<Concrete>())
-    , impl_(std::addressof(impl)) {}
+    , impl_((void*) std::addressof(impl)) {}
 
   void swap(type& other) noexcept {
     std::swap(vtable_, other.vtable_);
     std::swap(impl_, other.impl_);
   }
 
+  // Two any_ref's compare equal IFF they refer to the same object (shallow).
+  friend bool operator==(type const& left, type const& right) noexcept {
+    return left.impl_ == right.impl_;
```

Comment:
Do you also need to compare the vtables?
I'm just wondering about a case where you have an any_ref constructed from a ref-to-base and an any_ref constructed from ref-to-derived. These may have vtables that dispatch to different implementations and have different behaviour.

But then how do you compare vtables?
There is no guarantee that the function-pointers will be the same across different translation units.
I guess the vtable could store the type-id it was generated from as the first element and you could compare that. Might be a rabbit hole, though.

Do we really _need_ equality comparisons here?

Reply:
Copy and assign is shallow, so equality comparison should be shallow also. That's part of regularity. And I provide it because it's hard to implement as a CPO, but cheap and easy for the type to provide it directly.